aviatrix/utils.go

```diff
@@ -216,6 +216,18 @@ func DiffSuppressFuncGCPVpcId(k, old, new string, d *schema.ResourceData) bool {
 	return false
 }
 
+func DiffSuppressFuncNatInterface(k, old, new string, d *schema.ResourceData) bool {
+	connectionKey := strings.Replace(k, "interface", "connection", 1)
+	connection := d.Get(connectionKey).(string)
+	// Must check if ID != "" so that interface can be set on initial create. Without
+	// this check, the value for interface will be suppressed and interface = "" will
+	// be passed to the API even if interface = "eth0" in the configuration.
+	if d.Id() != "" && !(connection == "" || connection == "None") {
+		return old == "" && new == "eth0"
```

Comment:
The main cases to consider for this are the following

1. Policy-based connection - old should never be empty so this should not effect it
2. Route-based connection - old will always be empty so this will suppress a diff when new == "eth0"
3. No connection - No changes